modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -733,11 +733,18 @@ public List<Map<String, Serializable>> getValuesList(
 				)
 			).orderBy(
 				orderByExpressions
-			).limit(
-				start, end
 			),
 			selectExpressions);
 
+		rows = rows.stream(
+		).skip(
+			start
+		).limit(
+			end - start
+		).collect(
+			Collectors.toList()
+		);
+
```

Comment:
@quanhuynhces I don't think that paginating the results like this is good. The size of the results should be added to the query before executing it.
Could you please double-check if the `objectEntryPersistence.dslQuery()` function can be used instead of `_list()`?
Or adding the limit and offset to the query is also an option.
Thanks.